### PR TITLE
Document the trial, enterprise and community license behaviour.

### PR DESCRIPTION
**Pull request by andreidan**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: blackbox/docs/enterprise/index.rst [hunk 1]
```diff
@@ -4,9 +4,28 @@
...
+
+Alternatively, users seeking to use the community edition can do so at any
+point by disabling the enterprise license using the
+:ref:`license.enterprise <conf-node-enterprise-license>` config.
+
+.. WARNING::
+
+  CrateDB will continue to run after the license expires but only the ``SET``
+  statement and the read only system information statements (against the ``sys``
+  and ``information_schema`` schemas) will be allowed.
```
[the remark this answers is not in the document]
Comment (andreidan) [reply blackbox/docs/enterprise/index.rst]: I think we should specify that monitoring the crate cluster is still allowed

File: blackbox/docs/enterprise/index.rst [hunk 2]
```diff
@@ -4,9 +4,28 @@
...
+:ref:`license.enterprise <conf-node-enterprise-license>` config being true, for
+a period of 30 days that will allow users to evaluate the enterprise features.
+
+Upon the expiration of the 30 days trial period you will have to request an
+enterprise license by `contacting our sales department`_.
+Setting a new license is achieved using the :ref:`SET LICENSE <ref-set>`
+statement.
+
+Alternatively, users seeking to use the community edition can do so at any
+point by disabling the enterprise license using the
```
[the remark this answers is not in the document]
Comment (andreidan) [reply blackbox/docs/enterprise/index.rst]: the next line links to the configuration

File: blackbox/docs/enterprise/index.rst [hunk 3]
```diff
@@ -4,9 +4,28 @@
...
+:ref:`license.enterprise <conf-node-enterprise-license>` config.
+
+.. WARNING::
+
+  CrateDB will continue to run after the license expires but only the ``SET``
```
Comment (mfussenegger) [inline blackbox/docs/enterprise/index.rst]:
```suggestion
  CrateDB will continue to run after the license expires but only the ``SET LICENSE``
```

Or do we allow all SET statements?

File: blackbox/docs/enterprise/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfussenegger) [reply blackbox/docs/enterprise/index.rst]: I agree that we should mention it.

File: blackbox/docs/enterprise/index.rst (hunk 3 above)
Comment (andreidan) [reply blackbox/docs/enterprise/index.rst]: 👍 made the change (one `SET LICENSE` is allowed)

File: blackbox/docs/enterprise/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (smakalias) [reply blackbox/docs/enterprise/index.rst]: ah, sorry I missed this

**Pull request merged**